dht/krpc_hook.go

```diff
@@ -0,0 +1,14 @@
+package dht
+
+import "net"
+
+// A hook function for handling incoming KRPC messages.
+// Hooks can't pass errors back, so if error logging or
+// management within hooks is desired, consider passing
+// methods of a custom type as KRPCHooks and use the type
+// to handle errors.
+// The arguments are the incoming DHTAddr and Msg.
+// The outputs are:
+//  - Optional replacement Msg for ensuing use in default handlers
+//  - Bool indicating whether to propagate msg to default handlers.
+type KRPCHook func(*net.Addr, *Node, *Msg) (*Msg, bool)
```

Comment:
I think this file should be removed. OnQuery func(*Msg, net.Addr) bool is fine. Note that net.Addr is an interface type, so you don't need to make it a pointer.